### PR TITLE
feat: add tauri-plugin-network for network availability monitoring

### PR DESCRIPTION
**Pull request by yujonglee**
## Summary

Adds a new `tauri-plugin-network` plugin that monitors network availability using an actor-based approach. The plugin:

- Creates a `NetworkActor` that periodically checks network connectivity by fetching `https://www.google.com/generate_204` every 2 seconds
- Emits `NetworkStatusEvent` when network status changes (online ↔ offline)
- Links to the root supervisor when `parent_supervisor` is provided in `InitOptions`
- Uses `ractor::time::send_after` for scheduling periodic checks with minimal CPU/battery overhead

## Review & Testing Checklist for Human

- [ ] **Verify supervisor linking behavior**: The actor is spawned independently and then linked to the parent supervisor via `actor_ref.get_cell().link(parent_cell)`. This is different from spawning as a child of a DynamicSupervisor. Confirm this provides the expected supervision/restart behavior.
- [ ] **Check timing edge case**: Request timeout is 5s but check interval is 2s. If network is slow, could this cause overlapping requests? Consider if this needs adjustment.
- [ ] **Test event emission**: Run the app and verify `NetworkStatusEvent` is properly emitted when toggling network connectivity (e.g., disable WiFi)
- [ ] **Initial state**: Actor starts assuming `is_online: true`. First event only emits after first check if status differs. Verify this is acceptable behavior.

**Recommended test plan:**
1. Run `ONBOARDING=0 pnpm -F desktop tauri dev`
2. Add a listener for `NetworkStatusEvent` in the frontend (or check logs)
3. Toggle network connectivity and verify events are emitted on status change

### Notes
- No commands are exposed - only events. If you need to query current status, a command would need to be added.
- TypeScript bindings not generated yet - would need `pnpm -F @hypr/plugin-network codegen` after adding package.json/tsconfig.json

Requested by: @yujonglee (yujonglee.dev@gmail.com)
Link to Devin run: https://app.devin.ai/sessions/6cdf968dba9e41d3bde5687ff85c5c0b

### CODE REVIEW

File: plugins/network/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                    Err(e) => {
+                        tracing::error!(?e, "failed_to_spawn_network_actor");
+                    }
+                }
+            });
+
+            Ok(())
+        })
+        .build()
+}
```
[the remark this answers is not in the document]
Comment (yujonglee) [reply plugins/network/src/lib.rs:58]: @devin take a look
